.circleci/config.yml

```diff
@@ -106,61 +114,109 @@ jobs:
                     - composer-cache
 
             - run:
-                name: dependencies
-                command: ./.circleci/set-up-globals.sh
+                name: setup-environment-vars
+                command: /build-tools-ci/scripts/set-environment
+
+            - run:
+                name: log in
+                command: terminus -n auth:login --machine-token="$TERMINUS_TOKEN"
+
+            - run:
+                name: install dev dependencies, build assets, etc.
+                command: ./.ci/scripts/01-prepare
 
             - run:
                 name: build assets
                 command: composer -n build-assets
 
             - run:
-                name: deploy to Pantheon
-                command: ./.ci/deploy-to-pantheon.sh
+                name: prepare database for site-under test
+                command: ./.ci/scripts/02-init-site-under-test-clone-existing
+                # command: ./.ci/scripts/02-init-site-under-test-reinstall-new
 
-    behat_test:
-        <<: *defaults
-        steps:
-            - checkout
+            - run:
+                name: run composer install again to get dev dependencies
+                command: composer install
 
-            - attach_workspace:
-                # Must be absolute path or relative path from working_directory
-                at: /tmp
+            - run:
+                name: run functional tests with Behat
+                command: ./.ci/scripts/03-test
 
-            - restore_cache:
-                keys:
-                    - terminus-install
+            - run:
+                name: behat-cleanup
+                command: ./.ci/scripts/behat-cleanup
+                when: always
 
             - run:
-                name: copy vendor
-                command: cp -r /tmp/vendor .
-            
+                name: post-test actions
+                command: ./.ci/scripts/04-post-test
+
             - run:
-                name: dependencies
-                command: ./.circleci/set-up-globals.sh
+                name: handle merge to master (if needed)
```

Comment:
By calling `merge` from the functional step, is it possible that a merge to master/Pantheon dev happens even if other jobs, like `unit`, fail? I understand that `example-drops-8-composer` does the same thing. But yesterday I was trying to bring a personal project back in line with `example-drops-8-composer` and got hung up on this detail.

Reply:
@stevector I'm inclined to agree. I think the `functional` job does too much and could be split into multiple jobs.

A `merge-master` job could require both `unit` and `functional` test jobs, while still allowing those two to run in parallel.

I think there could even be a `deploy-dev` job that the `functional` test job requires.

This would be a change we make both here and in `example-drops-8-composer`. Are you okay with merging this as-is and making a follow-up JIRA issue for further discussion?

Reply:
Card `BT-42` created to track this discussion